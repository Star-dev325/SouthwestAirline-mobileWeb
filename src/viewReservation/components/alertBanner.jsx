// @flow
import React from 'react';
import _ from 'lodash';
import cx from 'classnames';
import Icon from 'src/shared/components/icon';
import i18n from '@swa-ui/locale';

type Props = {
  message: string,
  onClick?: () => void,
  hideRightArrow: boolean,
  important: boolean,
  hideAlertIcon?: boolean
};

const AlertBanner = ({ message, onClick = _.noop, hideRightArrow, important, hideAlertIcon }: Props) => (
  <div className="alert-banner" onClick={onClick}>
    <div className="alert-banner--content">
      <div>
        {!hideAlertIcon && <Icon type="travel-alert" className="larger alert-banner--vertically-centered-icon" />}
        <span className={cx(hideAlertIcon ? 'mx0' : 'mx7')}>
          {important && <b>{i18n('VIEW_RESERVATION__ALERT_BANNER__IMPORTANT')}</b>}
          {message}
        </span>
      </div>
      {!hideRightArrow && (
        <Icon
          type="keyboard-arrow-right"
          className="xxxlarge alert-banner--vertically-centered-icon alert-banner--arrow-icon"
        />
      )}
    </div>
  </div>
);

export default AlertBanner;
