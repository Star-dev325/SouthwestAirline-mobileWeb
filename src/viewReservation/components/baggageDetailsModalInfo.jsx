// @flow
import React from 'react';
import Icon from 'src/shared/components/icon';
import type { ModalInfoType } from 'src/shared/flow-typed/shared.types';

type Props = {
  modalInfo: ModalInfoType
};

const BaggageDetailsModalInfo = ({ modalInfo: { icon, text } }: Props) => (
  <div className="baggage-details--modal-info-content">
    <Icon className="baggage-details--modal-info-icon" type={icon} />
    <p className="baggage-details--modal-info-text">{text}</p>
  </div>
);

export default BaggageDetailsModalInfo;
