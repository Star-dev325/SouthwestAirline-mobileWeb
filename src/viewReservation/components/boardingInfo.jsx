// @flow
import { useHref } from '@swa-ui/encryption';
import i18n from '@swa-ui/locale';
import _ from 'lodash';
import React from 'react';
import { connect } from 'react-redux';
import CheckBaggageButton from 'src/checkIn/components/checkBaggageButton';
import MobileBoardingPassMessage from 'src/checkIn/components/mobileBoardingPassMessage';
import { hideDialog, showDialog } from 'src/shared/actions/dialogActions';
import { raiseSatelliteEvent } from 'src/shared/analytics/helpers/analyticsEventHelper';
import Button from 'src/shared/components/button';
import ButtonWithPlusIconAndText from 'src/shared/components/buttonWithPlusIconAndText';
import CheckInButton from 'src/shared/components/checkInButton';
import ConfirmationNumber from 'src/shared/components/confirmationNumber';
import Icon from 'src/shared/components/icon';
import TripDateAndDestCity from 'src/shared/components/tripDateAndDestCity';
import YellowButton from 'src/shared/components/yellowButton';
import { TRACK_CHECKED_BAGS, VIEW_MODIFY_CHECKED_BAGS } from 'src/shared/constants/checkedBagsFeatureNames';
import { getCheckedBagsQueryParams } from 'src/shared/helpers/checkedBagsQueryParams';
import DayOfTravelContactSelect from 'src/viewReservation/components/dayOfTravelContactSelect';
import DayOfTravelPassengerGroup from 'src/viewReservation/components/dayOfTravelPassengerGroup';
import { getShowDialogOptions } from 'src/viewReservation/helpers/viewReservationHelper';
import type { Node } from 'react';

import type { OptionsAndNextSteps } from 'src/myAccount/flow-typed/myAccount.types';
import type {
  CheckedBagsType,
  GreyBoxMessage,
  ModifyBaggageDetailsType,
  SameDayBlockedMessageType,
  SameDayReservation,
  ViewBoardingPass
} from 'src/shared/flow-typed/shared.types';
import type { Passenger } from 'src/viewReservation/components/passengerReservationInfo';
import type { FlightRetrieveRequestType } from 'src/viewReservation/flow-typed/viewReservation.types';

type Props = {
  checkIn: ?Link,
  checkInIneligibilityReason: ?string,
  confirmationNumber: string,
  contactInformation: ?Link,
  contactTracing: ?Link,
  date: string,
  dayOfTravelContactInfo: string,
  destinationAirport: string,
  destinationDescription: string,
  greyBoxMessage: ?GreyBoxMessage,
  greyBoxPassengerMessage: ?GreyBoxMessage,
  hasUnaccompaniedMinor: boolean,
  hideDialogFn: (*) => Promise<*>,
  isCheckInEligible: boolean,
  isInternational: boolean,
  modifyBaggageDetails: ModifyBaggageDetailsType,
  onAddCompanionButtonClick: () => void,
  onCancelFlightClick: () => void,
  onChangeFlightClick: () => void,
  onCheckBagsButtonClick: (string) => void,
  onCheckInButtonClick: (*) => void,
  onContactInfoClick: () => void,
  onContactTracingButtonClick: () => void,
  onEarlyBirdButtonClick: () => void,
  onPassengerNameClick: (string) => void,
  onSameDayButtonClick: (sameDayUpdates: ?SameDayReservation) => void,
  onViewBoardingPassButtonClickCb: (*, *) => void,
  onViewBoardingPositionsButtonClick: (*) => void,
  optionsAndNextSteps?: OptionsAndNextSteps,
  originAirport: string,
  passengers: Array<Passenger>,
  sameDayBlockedMessage: ?SameDayBlockedMessageType,
  sameDayUpdates: ?SameDayReservation,
  shouldShowAddCompanionButton: boolean,
  shouldShowAddEarlyBirdButton: boolean,
  shouldShowContactTracingButton: boolean,
  shouldSuppressUnmodifiablePnr: boolean,
  showDialogFn: (*) => Promise<*>,
  trackCheckedBags: CheckedBagsType,
  UI_ENCRYPTION: boolean,
  viewBoardingPass: ?ViewBoardingPass,
  viewBoardingPassIssuance: ?ViewBoardingPass,
  viewBoardingPositions: ?Link,
  viewModifyCheckedBags: CheckedBagsType,
  viewReservationSearchRequest: FlightRetrieveRequestType
};

export const BoardingInfo = (props: Props) => {
  const {
    checkIn,
    confirmationNumber,
    contactInformation,
    contactTracing = {},
    date,
    dayOfTravelContactInfo,
    destinationAirport,
    destinationDescription,
    greyBoxMessage,
    greyBoxPassengerMessage,
    hideDialogFn,
    isInternational,
    modifyBaggageDetails,
    onAddCompanionButtonClick,
    onCancelFlightClick,
    optionsAndNextSteps,
    onChangeFlightClick,
    onCheckBagsButtonClick,
    onCheckInButtonClick,
    onContactInfoClick,
    onContactTracingButtonClick,
    onEarlyBirdButtonClick,
    onPassengerNameClick,
    onSameDayButtonClick,
    onViewBoardingPassButtonClickCb,
    onViewBoardingPositionsButtonClick,
    originAirport,
    passengers,
    sameDayBlockedMessage,
    sameDayUpdates,
    shouldShowAddEarlyBirdButton,
    shouldShowContactTracingButton,
    shouldSuppressUnmodifiablePnr,
    showDialogFn,
    trackCheckedBags,
    UI_ENCRYPTION,
    viewBoardingPass,
    viewBoardingPassIssuance,
    viewBoardingPositions,
    viewModifyCheckedBags,
    viewReservationSearchRequest
  } = props;

  const {
    body,
    labelText: sameDayBlockedMessageLabelText,
    shouldShowCheckInButton,
    shouldShowModifyBagsButton
  } = sameDayBlockedMessage || {};
  const { firstName = '', lastName = '', recordLocator = '' } = viewReservationSearchRequest || {};

  const _renderAddEarlyBird = (): Node =>
    shouldShowAddEarlyBirdButton && (
      <Button
        className="boarding-info--early-bird-button"
        size="large"
        color="yellow"
        onClick={onEarlyBirdButtonClick}
        fluid
      >
        <Icon type="early-bird" color="yellow" />
        {i18n('SHARED__EARLY_BIRD__CHECK_IN_TITLE')}
      </Button>
    );

  const _sameDayUpdateAndBlockedClickFn = (checkBagsHref) => {
    if (sameDayBlockedMessage) {
      const label =
        shouldShowModifyBagsButton || shouldShowCheckInButton
          ? i18n('SHARED__BUTTON_TEXT__CANCEL')
          : i18n('SHARED__BUTTON_TEXT__OK');
      const buttons = [{ label, onClick: hideDialogFn }];

      if (shouldShowCheckInButton) {
        raiseSatelliteEvent('squid', { page_description: 'modal: sdc/sb not checked in' });
        buttons.push({
          label: i18n('VIEW_RESERVATION__BOARDING_INFO__CHECK_IN_BUTTON_TEXT'),
          onClick: () => {
            hideDialogFn().then(() => {
              onCheckInButtonClick(checkIn);
            });
          }
        });
      } else if (shouldShowModifyBagsButton) {
        buttons.push({
          label: i18n('VIEW_RESERVATION__BOARDING_INFO__REMOVE_BAGS_BUTTON_TEXT'),
          onClick: () => {
            hideDialogFn().then(() => {
              onCheckBagsButtonClick(checkBagsHref);
            });
          }
        });
      }

      showDialogFn({ name: 'sameDayBlockedMessage', message: body, buttons });
    } else {
      onSameDayButtonClick(sameDayUpdates);
    }
  };

  const _renderSameDayChangeButtons = () => {
    const { url } = viewModifyCheckedBags ?? {};
    let checkBagsHref = url;

    if (!sameDayUpdates && !sameDayBlockedMessage) return null;

    if (UI_ENCRYPTION) {
      const dataToEncrypt = {
        first_name: firstName,
        last_name: lastName,
        record_locator: recordLocator
      };
      const { href } = useHref(dataToEncrypt, url, '<BoardingInfo>', 'viewModifyCheckedBags');

      checkBagsHref = href;
    }

    return (
      <div className="boarding-info--change-cancel-container" data-qa="boarding-info-manage-buttons">
        <Button
          className="boarding-info--same-day-change-buttons"
          color="blue"
          fluid
          onClick={() => _sameDayUpdateAndBlockedClickFn(checkBagsHref)}
          size="large"
        >
          {sameDayUpdates?.labelText ?? sameDayBlockedMessageLabelText}
        </Button>
      </div>
    );
  };

  const _renderChangeOrCancelButtons = () => {
    if (shouldSuppressUnmodifiablePnr) {
      return null;
    }

    return (
      <div className="boarding-info--change-cancel-container" data-qa="boarding-info-manage-buttons">
        <Button
          className="manage-button-change boarding-info--change-cancel-buttons"
          color="blue"
          size="large"
          onClick={onChangeFlightClick}
          fluid
        >
          {i18n('VIEW_RESERVATION__BOARDING_INFO__CHANGE_BUTTON_TEXT')}
        </Button>
        <Button
          className="manage-button-cancel boarding-info--change-cancel-buttons"
          color="blue"
          size="large"
          onClick={onCancelFlightClick}
          fluid
        >
          {i18n('VIEW_RESERVATION__BOARDING_INFO__CANCEL_BUTTON_TEXT')}
        </Button>
      </div>
    );
  };

  const _renderCheckBagsNowButton = () => {
    const { query: { first_name = firstName, last_name = lastName, record_locator = recordLocator } = {} } =
      trackCheckedBags || {};

    const checkInRequest = {
      body: {
        firstName: first_name,
        lastName: last_name,
        recordLocator: record_locator
      }
    };

    return (
      <CheckBaggageButton
        buttonClassName="check-baggage-button-boarding"
        buttonSize="large"
        checkedBagsData={trackCheckedBags ? trackCheckedBags : viewModifyCheckedBags}
        checkInRequest={checkInRequest}
        component="<BoardingInfo>"
        feature={trackCheckedBags ? TRACK_CHECKED_BAGS : VIEW_MODIFY_CHECKED_BAGS}
        icon={trackCheckedBags ? 'ic-external-link' : null}
        queryParams={getCheckedBagsQueryParams(trackCheckedBags)}
      />
    );
  };

  const _renderBaggageDetails = () => {
    const { linkIcon, linkPrefixText, linkSuffixClickableText } = modifyBaggageDetails;

    const _handleOnSuffixTextClick = () => {
      showDialogFn(getShowDialogOptions(modifyBaggageDetails, hideDialogFn));
    };

    return (
      <div className="baggage-details">
        <Icon className="baggage-details--icon" type={linkIcon} />
        <div>
          <p>{linkPrefixText} </p>
          <p className="baggage-details--suffix-text" onClick={_handleOnSuffixTextClick}>
            {linkSuffixClickableText}
          </p>
        </div>
      </div>
    );
  };

  const _renderGreyBoxPnrMessage = (): Node => <MobileBoardingPassMessage greyBoxMessage={greyBoxMessage} />;

  const _renderGreyBoxPassengerMessage = (): Node => (
    <MobileBoardingPassMessage greyBoxMessage={greyBoxPassengerMessage} />
  );

  const _renderAddCompanionButton = (): Node => {
    let button = null;

    if (props.shouldShowAddCompanionButton) {
      button = (
        <ButtonWithPlusIconAndText onClick={onAddCompanionButtonClick} data-qa="add-companion">
          {i18n('VIEW_RESERVATION__BOARDING_INFO__ADD_COMPANION')}
        </ButtonWithPlusIconAndText>
      );
    }

    return button;
  };

  const passengerGroupsByCheckInStatus = _.groupBy(passengers, 'isCheckedIn');
  const passengersNotCheckedInGroupedByEligibility = _.groupBy(
    passengerGroupsByCheckInStatus['false'],
    'isCheckInEligible'
  );

  const _renderOptionsAndNextSteps= () => (
    <div>
      <div className="mt3">
        <YellowButton
          className="detailed-trip-card--options-btn"
          href={optionsAndNextSteps?.url}
          title={optionsAndNextSteps?.labelText}
        />
      </div>
    </div>
  );

  const _renderViewBoardingPassSection = () =>
    (!_.isEmpty(viewBoardingPass) || !_.isEmpty(viewBoardingPassIssuance)) && (
      <DayOfTravelPassengerGroup
        passengers={passengers}
        showPassengerHeader
        isInternational={isInternational}
        onPassengerNameClick={onPassengerNameClick}
      >
        <YellowButton
          title={_.get(viewBoardingPassIssuance, 'labelText', i18n('SHARED__BUTTON_TEXT__VIEW_BOARDING_PASS'))}
          className="mt4"
          onClick={() => onViewBoardingPassButtonClickCb(confirmationNumber)}
        />
      </DayOfTravelPassengerGroup>
    );

  const _renderDayOfTravelContactSelect = () => (
    <DayOfTravelContactSelect
      dayOfTravelContactInfo={dayOfTravelContactInfo}
      contactInformation={contactInformation}
      onContactInfoClick={onContactInfoClick}
    />
  );

  const _renderViewBoardingDetailsSection = () =>
    passengerGroupsByCheckInStatus[true] &&
    !_.isEmpty(viewBoardingPositions) && (
      <DayOfTravelPassengerGroup
        passengers={passengerGroupsByCheckInStatus[true]}
        showPassengerHeader
        isInternational={isInternational}
        onPassengerNameClick={onPassengerNameClick}
      >
        <YellowButton
          title={_.get(
            viewBoardingPositions,
            'labelText',
            !isInternational
              ? i18n('SHARED__BUTTON_TEXT__BOARDING_PASSES')
              : i18n('SHARED__BUTTON_TEXT__VIEW_BOARDING_DETAILS')
          )}
          className="mt4"
          onClick={() => onViewBoardingPositionsButtonClick(viewBoardingPositions)}
        />
      </DayOfTravelPassengerGroup>
    );

  const _renderCheckInButtonSection = () =>
    passengersNotCheckedInGroupedByEligibility[true] &&
    !_.isEmpty(checkIn) && (
      <DayOfTravelPassengerGroup
        passengers={passengersNotCheckedInGroupedByEligibility[true]}
        showPassengerHeader={
          _.isEmpty(viewBoardingPositions) && _.isEmpty(viewBoardingPass) && _.isEmpty(viewBoardingPassIssuance)
        }
        isInternational={isInternational}
        onPassengerNameClick={onPassengerNameClick}
      >
        <CheckInButton className="boarding-info--checkin-button" onClick={() => onCheckInButtonClick(checkIn)} />
      </DayOfTravelPassengerGroup>
    );

  const _renderNotEligibleForCheckInSection = () => {
    const ineligiblePax = _.isEmpty(checkIn)
      ? passengerGroupsByCheckInStatus['false']
      : passengersNotCheckedInGroupedByEligibility['false'];

    return (
      ineligiblePax && (
        <DayOfTravelPassengerGroup
          passengers={ineligiblePax}
          showPassengerHeader={
            _.isEmpty(checkIn) &&
            _.isEmpty(viewBoardingPositions) &&
            _.isEmpty(viewBoardingPass) &&
            _.isEmpty(viewBoardingPassIssuance)
          }
          isInternational={isInternational}
          onPassengerNameClick={onPassengerNameClick}
        />
      )
    );
  };

  return (
    <div className="trip-details-boarding-info">
      <div className="boarding-info">
        <div className="boarding-info_time reservation-details--card boarding-info--divider">
          <div className="flex flex-main-between">
            <TripDateAndDestCity date={date} cityName={destinationDescription} className="pblue" />
            <ConfirmationNumber confirmationNumber={confirmationNumber} />
          </div>
          <div className="airport-info--detail">
            {originAirport} to
            <br />
            {destinationAirport}
          </div>
          {_renderAddEarlyBird()}
          {_renderSameDayChangeButtons()}
          {optionsAndNextSteps && _renderOptionsAndNextSteps()}
          {_renderChangeOrCancelButtons()}
          {(viewModifyCheckedBags || trackCheckedBags) && _renderCheckBagsNowButton()}
          {modifyBaggageDetails && _renderBaggageDetails()}
          {greyBoxMessage && _renderGreyBoxPnrMessage()}
          {_renderAddCompanionButton()}
          {contactInformation && _renderDayOfTravelContactSelect()}
        </div>
        {_renderViewBoardingPassSection()}
        {_renderViewBoardingDetailsSection()}
        {_renderCheckInButtonSection()}
        {_renderNotEligibleForCheckInSection()}
        {greyBoxPassengerMessage && _renderGreyBoxPassengerMessage()}
        {shouldShowContactTracingButton && (
          <Button
            color="blue"
            size="larger"
            className="boarding-info--contact-tracing-button mt2"
            onClick={onContactTracingButtonClick}
            fluid
          >
            {_.get(contactTracing, 'labelText')}
          </Button>
        )}
      </div>
    </div>
  );
};

BoardingInfo.defaultProps = {
  onPassengerNameClick: _.noop,
  onEarlyBirdButtonClick: _.noop,
  cancelFlight: _.noop,
  onCheckInButtonClick: _.noop
};

const mapStateToProps = () => ({});

const mapDispatchToProps = {
  showDialogFn: showDialog,
  hideDialogFn: hideDialog
};

export default _.flowRight(connect(mapStateToProps, mapDispatchToProps))(BoardingInfo);
