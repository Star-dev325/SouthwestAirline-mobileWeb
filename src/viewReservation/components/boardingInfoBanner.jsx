// @flow
import i18n from '@swa-ui/locale';
import _ from 'lodash';
import React from 'react';
import { REACCOM_VIEW_RESERVATION_MESSAGE } from 'src/airChange/constants/airChangeConstants';
import withFeatureToggles from 'src/shared/enhancers/withFeatureToggles';
import AlertBanner from 'src/viewReservation/components/alertBanner';
import ReaccomBanner from 'src/viewReservation/components/reaccomBanner';
import type { MessageType } from 'src/shared/flow-typed/shared.types';

const {
  REACCOM_CONTACT_US_TO_CHANGE_FLIGHT_FREE,
  REACCOM_VIEW_RESERVATION,
  REACCOM_VIEW_RESERVATION_FLIX,
  REACCOM_VIEW_RESERVATION_GDS
} = REACCOM_VIEW_RESERVATION_MESSAGE;

type Props = {
  isDynamicWaiver: boolean,
  messages: Array<MessageType>,
  hasAnyCancelledFlights: boolean,
  onClick: () => void,
  isChangeLinkEmpty?: boolean,
  isCancelLinkEmpty?: boolean,
  important: boolean,
  toggles?: *
};

export const BoardingInfoBanner = ({
  isDynamicWaiver,
  hasAnyCancelledFlights,
  isCancelLinkEmpty,
  isChangeLinkEmpty,
  onClick,
  important,
  messages
}: Props) => {
  const reaccomMessage =
    messages &&
    messages.find((message) => {
      const { key } = message;

      return [
        REACCOM_CONTACT_US_TO_CHANGE_FLIGHT_FREE,
        REACCOM_VIEW_RESERVATION,
        REACCOM_VIEW_RESERVATION_FLIX,
        REACCOM_VIEW_RESERVATION_GDS
      ].includes(key);
    });
  const reaccomBannerProps = {
    body: _.get(reaccomMessage, 'body'),
    header: _.get(reaccomMessage, 'header'),
    showBodyAsHtml: reaccomMessage && reaccomMessage.key === REACCOM_CONTACT_US_TO_CHANGE_FLIGHT_FREE
  };
  const alertBannerProps = {
    message: hasAnyCancelledFlights
      ? i18n('VIEW_RESERVATION__RESERVATION_DETAILS__FLIGHT_CANCELLED')
      : i18n('VIEW_RESERVATION__RESERVATION_DETAILS__YOU_CAN_CHANGE_YOUR_FLIGHT'),
    onClick,
    hideRightArrow: hasAnyCancelledFlights,
    important
  };
  const noChangeOrCancelLinks = isChangeLinkEmpty || isCancelLinkEmpty;
  const dynamicOrCancelled = isDynamicWaiver || hasAnyCancelledFlights;

  if (reaccomMessage) {
    if (noChangeOrCancelLinks && isDynamicWaiver) {
      return (
        <div>
          <ReaccomBanner {...reaccomBannerProps} />
          <AlertBanner
            important={false}
            message={i18n('VIEW_RESERVATION__RESERVATION_DETAILS__GENERIC_TRAVEL_ADVISORY')}
            hideRightArrow
            hideAlertIcon
          />
        </div>
      );
    } else {
      return <ReaccomBanner {...reaccomBannerProps} />;
    }
  } else if (noChangeOrCancelLinks && isDynamicWaiver) {
    return (
      <AlertBanner
        important={false}
        message={i18n('VIEW_RESERVATION__RESERVATION_DETAILS__GENERIC_TRAVEL_ADVISORY')}
        hideRightArrow
        hideAlertIcon
      />
    );
  } else if (dynamicOrCancelled && !isChangeLinkEmpty) {
    return <AlertBanner {...alertBannerProps} />;
  } else {
    return null;
  }
};

BoardingInfoBanner.defaultProps = {
  isCancelLinkEmpty: false
};

export default withFeatureToggles(BoardingInfoBanner);
