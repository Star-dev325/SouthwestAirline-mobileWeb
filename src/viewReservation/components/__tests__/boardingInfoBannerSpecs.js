import { mount } from 'enzyme';
import React from 'react';
import { sandbox } from 'sinon';
import { REACCOM_VIEW_RESERVATION_MESSAGE } from 'src/airChange/constants/airChangeConstants';
import { BoardingInfoBanner } from 'src/viewReservation/components/boardingInfoBanner';
import { click } from 'test/unit/helpers/enzymeFormTestUtils';

const sinon = sandbox.create();

describe('boardingInfoBanner', () => {
  afterEach(() => {
    sinon.restore();
  });

  it('should display generic message when isCancelLinkEmpty is true', () => {
    const instance = render({ isCancelLinkEmpty: true, important: false, hideRightArrow: true });

    expect(instance.find('.alert-banner--arrow-icon')).to.not.exist;
    expect(instance.find('.alert-banner--vertically-centered-icon')).to.not.exist;
    expect(instance).to.contain.text('Travel Advisory in effect');
  });

  it('should display generic message when isChangeLinkEmpty is true', () => {
    const instance = render({ isChangeLinkEmpty: true, important: false, hideRightArrow: true });

    expect(instance.find('.alert-banner--arrow-icon')).to.not.exist;
    expect(instance.find('.alert-banner--vertically-centered-icon')).to.not.exist;
    expect(instance).to.contain.text('Travel Advisory in effect');
  });

  it('should not show reaccom banner when flight is not reaccom', () => {
    const instance = render();

    expect(instance.find('.reaccom-banner')).to.not.exist;
  });

  it('should not be clickable when flight is non-rev PNR', () => {
    const onClickStub = sinon.stub();
    const instance = render({ isNonRevPnr: true });

    click(instance.find('.alert-banner'));
    expect(onClickStub).not.have.been.called;
  });

  it('should display Dynamic Waiver message with right arrow for domestic flight', () => {
    const instance = render({ isDynamicWaiver: true, hideRightArrow: false });

    expect(instance).to.contain.text('You can change your flight');
    expect(instance.find('.alert-banner--arrow-icon')).to.exist;
  });

  it('should display flight cancel message without right arrow', () => {
    const instance = render({ hasAnyCancelledFlights: true });

    expect(instance).to.contain.text('Your flight has been cancelled');
    expect(instance.find('.icon_keyboard-arrow-right')).to.not.exist;
  });

  it('should not display Dynamic Waiver message if change link empty', () => {
    const instance = render({ isDynamicWaiver: true, hasAnyCancelledFlights: true, isChangeLinkEmpty: true });

    expect(instance.find('.icon_keyboard-arrow-right')).to.not.exist;
  });

  it('should clickable when flight is dynamic waiver', () => {
    const onClickStub = sinon.stub();
    const instance = render({ hasAnyCancelledFlights: true, onClick: onClickStub });

    click(instance.find('.alert-banner'));
    expect(onClickStub).have.been.called;
  });

  it('should clickable when flight is cancelled', () => {
    const onClickStub = sinon.stub();
    const instance = render({ isDynamicWaiver: true, onClick: onClickStub });

    click(instance.find('.alert-banner'));
    expect(onClickStub).have.been.called;
  });

  context('is Reaccom', () => {
    it('should display reaccom message when flight is reaccom', () => {
      const instance = render({
        isDynamicWaiver: false,
        messages: [
          {
            key: 'REACCOM_VIEW_RESERVATION',
            icon: '',
            body: 'reaccomBody',
            header: 'reaccomHeader'
          }
        ]
      });

      expect(instance.find('.reaccom-banner')).to.exist;
      expect(instance.find('.reaccom-banner-container--header')).to.contain.text('reaccomHeader');
      expect(instance.find('.reaccom-banner--body')).to.contain.text('reaccomBody');
    });

    it('should display reaccom message when reaccom link is null but contact us message is there', () => {
      const instance = render({
        isDynamicWaiver: false,
        messages: [
          {
            key: 'REACCOM_CONTACT_US_TO_CHANGE_FLIGHT_FREE',
            icon: '',
            body: 'reaccomBody',
            header: 'reaccomHeader'
          }
        ]
      });

      expect(instance.find('.reaccom-banner')).to.exist;
      expect(instance.find('.reaccom-banner-container--header')).to.contain.text('reaccomHeader');
      expect(instance.find('.reaccom-banner--body')).to.contain.text('reaccomBody');
    });

    it('should not display reaccom messaging when the flight is not reaccomed but the toggle is enabled', () => {
      const instance = render({
        isChangeLinkEmpty: true,
        isDynamicWaiver: false,
        important: false,
        hideRightArrow: true
      });

      expect(instance.find('.reaccom-banner')).to.not.exist;
    });

    it('should display travel advisory banner when reaccomed flight falls under dynamic waiver', () => {
      const instance = render({
        isChangeLinkEmpty: true,
        isCancelLinkEmpty: true,
        isDynamicWaiver: true,
        messages: [
          {
            key: 'REACCOM_VIEW_RESERVATION',
            icon: '',
            body: 'reaccomBody',
            header: 'reaccomHeader'
          }
        ]
      });

      expect(instance.find('.reaccom-banner')).to.exist;
      expect(instance.find('.alert-banner')).to.exist;
    });

    it('should display flight disruption warning message for flight which has been disrupted involuntarily (re-accommodated)', () => {
      const instance = render({
        isChangeLinkEmpty: true,
        isCancelLinkEmpty: true,
        isDynamicWaiver: true,
        messages: [
          {
            key: 'REACCOM_VIEW_RESERVATION_GDS',
            icon: 'WARNING',
            body: 'A disruption caused changes to your trip. If you need to make an adjustment to the itinerary',
            header: 'Your trip was modified.'
          }
        ]
      });
  
      expect(instance).toMatchSnapshot();
    });

    it('should display flight disruption warning with FLIX message for flight which has been cancelled but not re-booked', () => {
      const instance = render({
        isCancelLinkEmpty: true,
        isChangeLinkEmpty: true,
        isDynamicWaiver: true,
        messages: [
          {
            body: 'A change was made to your original reservation. We are currently working to rebook your flight, or you can also rebook for free, up to two times.',
            header: "We're sorry, we had to change your flight(s).",
            icon: 'WARNING',
            key: REACCOM_VIEW_RESERVATION_MESSAGE.REACCOM_VIEW_RESERVATION_FLIX,
            textColor: 'DEFAULT'
          }
        ]
      });

      expect(instance).toMatchSnapshot();
    });
  });

  const render = (props) => {
    const defaultProps = {
      isNonRevPnr: false,
      isDynamicWaiver: true,
      hasAnyCancelledFlights: false,
      onClick: sinon.stub(),
      important: true,
      messages: [{ key: '', icon: '', body: '', header: '' }]
    };

    return mount(<BoardingInfoBanner {...defaultProps} {...props} />);
  };
});
