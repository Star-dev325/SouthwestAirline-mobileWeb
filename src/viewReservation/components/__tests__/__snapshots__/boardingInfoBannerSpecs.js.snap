// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boardingInfoBanner is Reaccom should display flight disruption warning message for flight which has been disrupted involuntarily (re-accommodated) 1`] = `
<BoardingInfoBanner
  hasAnyCancelledFlights={false}
  important={true}
  isCancelLinkEmpty={true}
  isChangeLinkEmpty={true}
  isDynamicWaiver={true}
  isNonRevPnr={false}
  messages={
    Array [
      Object {
        "body": "A disruption caused changes to your trip. If you need to make an adjustment to the itinerary",
        "header": "Your trip was modified.",
        "icon": "WARNING",
        "key": "REACCOM_VIEW_RESERVATION_GDS",
      },
    ]
  }
  onClick={[Function]}
>
  <div>
    <ReaccomBanner
      body="A disruption caused changes to your trip. If you need to make an adjustment to the itinerary"
      header="Your trip was modified."
      showBodyAsHtml={false}
    >
      <div
        className="reaccom-banner"
      >
        <div
          className="reaccom-banner-container flex"
        >
          <div
            className="reaccom-banner-container--alert"
          >
            <Icon
              className="reaccom-banner-container--alert-icon"
              type="plus-icon"
            >
              <i
                className="reaccom-banner-container--alert-icon icon icon_plus-icon"
              >
                !
              </i>
            </Icon>
          </div>
          <b
            className="reaccom-banner-container--header"
          >
            Your trip was modified.
          </b>
        </div>
        <p
          className="reaccom-banner--body py2"
        >
          A disruption caused changes to your trip. If you need to make an adjustment to the itinerary
        </p>
      </div>
    </ReaccomBanner>
    <AlertBanner
      hideAlertIcon={true}
      hideRightArrow={true}
      important={false}
      message="Travel Advisory in effect"
    >
      <div
        className="alert-banner"
        onClick={[Function]}
      >
        <div
          className="alert-banner--content"
        >
          <div>
            <span
              className="mx0"
            >
              Travel Advisory in effect
            </span>
          </div>
        </div>
      </div>
    </AlertBanner>
  </div>
</BoardingInfoBanner>
`;

exports[`boardingInfoBanner is Reaccom should display flight disruption warning with FLIX message for flight which has been cancelled but not re-booked 1`] = `
<BoardingInfoBanner
  hasAnyCancelledFlights={false}
  important={true}
  isCancelLinkEmpty={true}
  isChangeLinkEmpty={true}
  isDynamicWaiver={true}
  isNonRevPnr={false}
  messages={
    Array [
      Object {
        "body": "A change was made to your original reservation. We are currently working to rebook your flight, or you can also rebook for free, up to two times.",
        "header": "We're sorry, we had to change your flight(s).",
        "icon": "WARNING",
        "key": "REACCOM_VIEW_RESERVATION_FLIX",
        "textColor": "DEFAULT",
      },
    ]
  }
  onClick={[Function]}
>
  <div>
    <ReaccomBanner
      body="A change was made to your original reservation. We are currently working to rebook your flight, or you can also rebook for free, up to two times."
      header="We're sorry, we had to change your flight(s)."
      showBodyAsHtml={false}
    >
      <div
        className="reaccom-banner"
      >
        <div
          className="reaccom-banner-container flex"
        >
          <div
            className="reaccom-banner-container--alert"
          >
            <Icon
              className="reaccom-banner-container--alert-icon"
              type="plus-icon"
            >
              <i
                className="reaccom-banner-container--alert-icon icon icon_plus-icon"
              >
                !
              </i>
            </Icon>
          </div>
          <b
            className="reaccom-banner-container--header"
          >
            We're sorry, we had to change your flight(s).
          </b>
        </div>
        <p
          className="reaccom-banner--body py2"
        >
          A change was made to your original reservation. We are currently working to rebook your flight, or you can also rebook for free, up to two times.
        </p>
      </div>
    </ReaccomBanner>
    <AlertBanner
      hideAlertIcon={true}
      hideRightArrow={true}
      important={false}
      message="Travel Advisory in effect"
    >
      <div
        className="alert-banner"
        onClick={[Function]}
      >
        <div
          className="alert-banner--content"
        >
          <div>
            <span
              className="mx0"
            >
              Travel Advisory in effect
            </span>
          </div>
        </div>
      </div>
    </AlertBanner>
  </div>
</BoardingInfoBanner>
`;
